theseus/src/error.rs

```diff
@@ -1,17 +1,25 @@
 //! Theseus error type
+use tracing_error::InstrumentError;
+
 #[derive(thiserror::Error, Debug)]
-pub enum Error {
+pub enum ErrorKind {
```

Comment:
What's the purpose of this name change? Most Rust libs use `Error`, and I think we should stick to that

Reply:
It is now integrated into the actual error type, which captures it and a spantrace so it can be traced back further than where application code is invoked.

Reply:
If there were a better solution for async, the authors of the Tokio runtime probably would have used it as the main way to trace errors in their own runtime (after all, Tokio is the one who authored the tracing library and is using it in their console debugger).

Reply:
IMO the old Error should be called `Error` because it's the direct library error. The other error should be `TracableError` or something of the sorts

Reply:
Actually, everywhere uses this kind of convention, backtraces are captured using a similar pattern usually.